Update ecommerce tracking in etracker

ben-dw opened the pull request:
I've added two more functions:

1. transaction (sale/lead/cancellation)
2. add to cart

Would like to hear feedback and the merge :)

ben-dw committed:
Update etracker for eCommerce transaction

ben-dw committed:
Update EtrackerTag.php

ben-dw committed:
Update EtrackerTag.php

ben-dw committed:
Update EtrackerTag.php

ben-dw committed:
Update EtrackerTag.php

ben-dw committed:
Update EtrackerTag.web.js

ben-dw committed:
Update EtrackerTag.php

Update Debug Mode

ben-dw committed:
Update EtrackerTag.web.js

Debug Mode

ben-dw committed:
Update EtrackerTag.web.js

ben-dw committed:
Update EtrackerTag.php

ben-dw committed:
Update EtrackerTag.php

ben-dw committed:
Update EtrackerTag.web.js

ben-dw committed:
Update EtrackerTag.php

ben-dw committed:
Update EtrackerTag.web.js

ben-dw committed:
Update EtrackerTag.php

ben-dw committed:
Update EtrackerTag.php

ben-dw committed:
Update EtrackerTag.web.js

ben-dw committed:
Update EtrackerTag.php

ben-dw committed:
Update EtrackerTag.web.js

ben-dw committed:
Update EtrackerTag.web.js

ben-dw committed:
Update EtrackerTag.php

ben-dw committed:
Update EtrackerTag.web.js

ben-dw committed:
Update EtrackerTag.web.js

ben-dw committed:
Update EtrackerTag.php

ben-dw committed:
Update EtrackerTag.php

ben-dw committed:
Update EtrackerTag.web.js

ben-dw commented:
Hello everybody, how can we get this thing into the next update?
Best, Ben

AltamashShaikh commented:
> Hello everybody, how can we get this thing into the next update? Best, Ben

@ben-dw Will try to review this soon and can merge this if there is no issue.

AltamashShaikh commented on Template/Tag/EtrackerTag.php:
```diff
@@ -121,6 +126,129 @@ public function getParameters()
                 $field->description = 'The event\'s type Name, for example an value of a send form...';
                 $field->customUiControlTemplateFile = self::FIELD_TEMPLATE_VARIABLE;
```
@ben-dw This has changed, and I am afraid you need to make the similar change in your code
Refer: https://github.com/matomo-org/tag-manager/blob/4.x-dev/Template/Tag/EtrackerTag.php#L41

AltamashShaikh commented on Template/Tag/EtrackerTag.php:
```diff
@@ -39,7 +44,7 @@ public function getParameters()
                 $field->description = 'Assign a etracker configuration in order to track data into a specific site.';
```
@ben-dw We have also moved the titles and description in `en.json` so that this could be translated.
Can you merge the latest code in your branch and move the `title` and `description` hard coded into `lang/en.json` and use `Piwik::translate()`, Refer this links 

https://github.com/matomo-org/tag-manager/blob/4.x-dev/Template/Tag/EtrackerTag.php#L26

https://github.com/matomo-org/tag-manager/blob/4.x-dev/lang/en.json#L704

AltamashShaikh commented on Template/Tag/EtrackerTag.web.js:
```diff
@@ -80,6 +80,25 @@
+                etCommerce.debugMode = true;
+            }
+            var etorder = {orderNumber:parameters.get('etrackerTransactionID'),status:parameters.get('etrackerTransactionType'),orderPrice:parameters.get('etrackerTransactionValue').toString(),basket:parameters.get('etrackerTransactionBasket'),currency:parameters.get('etrackerTransactionCurrency'),customerGroup:parameters.get('etrackerTransactionCustomerGroup'),deliveryConditions:parameters.get(' ...
+            etCommerce.sendEvent('order', etorder) ;
```
Why is this different from above code ` _etracker.sendEvent` ?

ben-dw replied:
This is another function, with another structure, only for ecommerce events.

AltamashShaikh reviewed (commented):
@ben-dw Left few comments and questions, overall the change looks like it could be merged easily.
Once the above queries are resolved, I can try to do a functional test.

ben-dw committed:
Merge branch 'matomo-org:4.x-dev' into ben-dw-etracker-update

ben-dw committed:
Added new etracker ecommerce translations

ben-dw committed:
Moved title and descriptions to lang/en.json

ben-dw committed:
Update EtrackerTag.web.js

Added async functions to ecommerce events.
"typeof(_etracker) === "object"" is still available.

ben-dw commented:
@AltamashShaikh Thank you for your support. I have updated translations and another ecommerce function.
I think, you can do a functional test :)

AltamashShaikh commented:
@ben-dw Looks good :tada: , Thanks for the PR :+1:
